frame/balances/ring/src/lib.rs

```diff
@@ -291,9 +297,9 @@ decl_storage! {
 					"the balance of any account should always be more than existential deposit.",
 				)
 			}
-			for &(ref who, free_ring) in config.balances.iter() {
+			for &(ref who, free) in config.balances.iter() {
 				let mut account_data = T::AccountStore::get(who);
-				account_data.free_ring = free_ring;
+				account_data.mutate_free(free);
```

Comment:
Review this: why compile succucess

Reply:
```^^^^^ cannot infer type for type parameter `T` declared on the trait `Data````

when multi trait impl

Reply:
A simulation, it seems that this will works:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=3be99c17bcc5244d75507d273ac7ea67